From 151d9535400afbe78d2905019e36c8f1bdefc954 Mon Sep 17 00:00:00 2001
From: Nicolas Aguirre <aguirre.nicolas@gmail.com>
Date: Fri, 9 Feb 2024 22:47:03 +0100
Subject: [PATCH 2/6] fix: remove double semi colons

---
 arch/arm/mach-sunxi/board.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm/mach-sunxi/board.c b/arch/arm/mach-sunxi/board.c
index 512d40a09b..9b0d68a7a0 100644
--- a/arch/arm/mach-sunxi/board.c
+++ b/arch/arm/mach-sunxi/board.c
@@ -230,7 +230,7 @@ static int suniv_get_boot_source(void)
 	}
 	/* If we get here something went wrong try to boot from FEL.*/
 	printf("Unknown boot source from BROM: 0x%x\n", brom_call);
-	return SUNXI_INVALID_BOOT_SOURCE;;
+	return SUNXI_INVALID_BOOT_SOURCE;
 }
 
 static int sunxi_egon_valid(struct boot_file_head *egon_head)
-- 
2.39.2

