From fcb55328cefd95c6c0111deae9d76cf635a609f9 Mon Sep 17 00:00:00 2001
From: Nicolas Aguirre <aguirre.nicolas@gmail.com>
Date: Sun, 21 Jan 2024 09:41:58 +0100
Subject: [PATCH 1/7] add support for DMA and SPDIF in f1c200s

---
 .../arm/boot/dts/allwinner/suniv-f1c100s.dtsi | 27 +++++++++++++++++++
 1 file changed, 27 insertions(+)

diff --git a/arch/arm/boot/dts/allwinner/suniv-f1c100s.dtsi b/arch/arm/boot/dts/allwinner/suniv-f1c100s.dtsi
index 3c61d59ab..7df350a9c 100644
--- a/arch/arm/boot/dts/allwinner/suniv-f1c100s.dtsi
+++ b/arch/arm/boot/dts/allwinner/suniv-f1c100s.dtsi
@@ -69,6 +69,27 @@ otg_sram: sram-section@0 {
 			};
 		};
 
+		dma: dma-controller@1c02000 {
+			compatible = "allwinner,sun4i-a10-dma";
+			reg = <0x01c02000 0x1000>;
+			interrupts = <18>;
+			clocks = <&ccu CLK_BUS_DMA>;
+			#dma-cells = <2>;
+		};
+
+		spdif: spdif@1c21400 {
+			#sound-dai-cells = <0>;
+			compatible = "allwinner,sun4i-a10-spdif";
+			reg = <0x01c21400 0x400>;
+			interrupts = <5>;
+			clocks = <&ccu CLK_SPDIF>;
+			clock-names = "apb", "spdif";
+			dmas = <&dma SUN4I_DMA_NORMAL 2>,
+			       <&dma SUN4I_DMA_NORMAL 2>;
+			dma-names = "rx", "tx";
+			status = "disabled";
+		};
+
 		spi0: spi@1c05000 {
 			compatible = "allwinner,suniv-f1c100s-spi",
 				     "allwinner,sun8i-h3-spi";
@@ -198,6 +219,12 @@ i2c0_pd_pins: i2c0-pd-pins {
 				function = "i2c0";
 			};
 
+			/omit-if-no-ref/
+			spdif_tx_pin: spdif-tx-pin {
+				pins = "PD17", "PE6";
+				function = "spdif";
+			};
+
 			spi0_pc_pins: spi0-pc-pins {
 				pins = "PC0", "PC1", "PC2", "PC3";
 				function = "spi0";
-- 
2.39.2

