From c7699f18b271f9765d48ad08b495d4db1c9fc335 Mon Sep 17 00:00:00 2001
From: Nicolas Aguirre <aguirre.nicolas@gmail.com>
Date: Tue, 28 May 2024 21:31:31 +0000
Subject: [PATCH 4/8] Add support for suniv spi

---
 drivers/spi/spi-sun6i.c | 15 +++++++++++++--
 1 file changed, 13 insertions(+), 2 deletions(-)

diff --git a/drivers/spi/spi-sun6i.c b/drivers/spi/spi-sun6i.c
index fddc63309..019ba8ada 100644
--- a/drivers/spi/spi-sun6i.c
+++ b/drivers/spi/spi-sun6i.c
@@ -94,6 +94,7 @@ struct sun6i_spi_cfg {
 	unsigned long		fifo_depth;
 	bool			has_clk_ctl;
 	u32			mode_bits;
+	u32			maxburst;
 };
 
 struct sun6i_spi {
@@ -220,7 +221,7 @@ static int sun6i_spi_prepare_dma(struct sun6i_spi *sspi,
 			.direction = DMA_DEV_TO_MEM,
 			.src_addr = sspi->dma_addr_rx,
 			.src_addr_width = DMA_SLAVE_BUSWIDTH_1_BYTE,
-			.src_maxburst = 8,
+			.src_maxburst = sspi->cfg->maxburst,
 		};
 
 		dmaengine_slave_config(master->dma_rx, &rxconf);
@@ -242,7 +243,8 @@ static int sun6i_spi_prepare_dma(struct sun6i_spi *sspi,
 			.direction = DMA_MEM_TO_DEV,
 			.dst_addr = sspi->dma_addr_tx,
 			.dst_addr_width = DMA_SLAVE_BUSWIDTH_4_BYTES,
-			.dst_maxburst = 8,
+			.dst_maxburst = sspi->cfg->maxburst,
+
 		};
 
 		dmaengine_slave_config(master->dma_tx, &txconf);
@@ -779,11 +781,13 @@ static void sun6i_spi_remove(struct platform_device *pdev)
 static const struct sun6i_spi_cfg sun6i_a31_spi_cfg = {
 	.fifo_depth	= SUN6I_FIFO_DEPTH,
 	.has_clk_ctl	= true,
+	.maxburst	= 8,
 };
 
 static const struct sun6i_spi_cfg sun8i_h3_spi_cfg = {
 	.fifo_depth	= SUN8I_FIFO_DEPTH,
 	.has_clk_ctl	= true,
+	.maxburst	= 8,
 };
 
 static const struct sun6i_spi_cfg sun50i_r329_spi_cfg = {
@@ -791,6 +795,12 @@ static const struct sun6i_spi_cfg sun50i_r329_spi_cfg = {
 	.mode_bits	= SPI_RX_DUAL | SPI_TX_DUAL | SPI_RX_QUAD | SPI_TX_QUAD,
 };
 
+static const struct sun6i_spi_cfg suniv_f1c100s_spi_cfg = {
+	.fifo_depth	= SUN6I_FIFO_DEPTH,
+	.has_clk_ctl	= true,
+	.maxburst	= 4,
+};
+
 static const struct of_device_id sun6i_spi_match[] = {
 	{ .compatible = "allwinner,sun6i-a31-spi", .data = &sun6i_a31_spi_cfg },
 	{ .compatible = "allwinner,sun8i-h3-spi",  .data = &sun8i_h3_spi_cfg },
@@ -798,6 +808,7 @@ static const struct of_device_id sun6i_spi_match[] = {
 		.compatible = "allwinner,sun50i-r329-spi",
 		.data = &sun50i_r329_spi_cfg
 	},
+	{ .compatible = "allwinner,suniv-f1c100s-spi",  .data = &suniv_f1c100s_spi_cfg },
 	{}
 };
 MODULE_DEVICE_TABLE(of, sun6i_spi_match);
-- 
2.39.2

